alchemist-loading/src/main/kotlin/it/unibo/alchemist/boundary/variables/SystemEnvVariable.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2010-2025, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.variables
+
+import it.unibo.alchemist.boundary.Variable
+import java.io.Serializable
+import java.util.stream.Stream
+
+/**
+ * A variable that retrieves its value from a system environment variable.
+ *
+ * @param name the name of the environment variable
+ * @param defaultValue the default value to return if the environment variable is not set
+ */
+class SystemEnvVariable(private val name: String, private val defaultValue: Serializable?) : Variable<Serializable> {
+    constructor(name: String) : this(name, null)
```

Comment:
```suggestion
/**
 * A variable that retrieves its value from a system environment variable.
 *
 * @param name the name of the environment variable
 * @param defaultValue the default value to return if the environment variable is not set.
 * If no default is provided and the environment variable is unset, this variable will throw an `IllegalStateException`
 */
@JvmOverloads
class SystemEnvVariable(private val name: String, private val defaultValue: Serializable?) : Variable<Serializable> {
```

Prefer default parameters + `@JvmOverloads` over constructor overloading